docs/DeveloperGuide.md

```diff
@@ -171,6 +171,46 @@ As such, the detailed descriptions for the Address Book subsystem above can be t
 
 This section describes some noteworthy details on how certain features are implemented.
 
+### Tag Management
+
+This section will describe tag management in the address book as well as the features implemented.
+
+### Centralising Tags in the Address Book
+
+In the previous implementation, all `Tag`s are independent of one another despite having the same tag names. Hence, centralising the tags enable the user to easily manipulate them as well as easily searching for `Person`s that contain the tag.
+This is done by creating a `UniqueTagList` within `AddressBook` which will store all tags that were created by the user. Whenever a command relating to `Tag` is executed, it will not only apply the changes to the `Tag` in the `UniqueTagList` but will propagate these changes to the relevant `Person`s who contain the specified tag.
+Hence, all operations relating to the `Tag`s are done at the `AddressBook` level to ensure that the `Tag`s and `Person`s are properly synchronised.
+
+Another benefit that comes with the centralised tag list is that the user can maintain tags even if it is not associated to any `Person`s. The rationale to maintain `Tag` is to allow the user to reuse the tag depending on their workflow (i.e. A user may want to maintain the `prospective clients` tag even if he/she currently does not have any prospective clients.)
+
+### Edit Tag Feature - `edittag`
+
+The tag editing feature is similar to the system used for `Person` but extended to propagate the changes to the `Person`s. This feature is implemented at the `AddressBook` level, and the related functions are:
+
+* `AddressBook#setTag(target, editedTag)`
+* `AddressBook#setPersonsWithTag(target, editedTag)`
+
+Note: `target` refers to the tag to be updated, and `editedTag` is the replacement tag specified by the user.
+
+![Tag Edit](images/TagEditSequenceDiagram.png)
+
+### Serialisation and Inflation
+
+`Tag` serialisation and inflation is handled by the `Storage` component. The current implementation extends the existing method from `JsonSerializableAddressBook` to read a list of tag names from the JSON file and saving them as its own JSON file.
```

Comment:
I understand what you mean when you say extend, but technically extend should refer to sub-classing (which you are not actually doing). Maybe this phrasing is better?
```suggestion
`Tag` serialisation and inflation is handled by the `Storage` component. The current implementation augments the existing method from `JsonSerializableAddressBook` through the addition of reading a list of tag names from the JSON file and saving them as its own JSON file.
```

Reply:
Fixed in 61207721